src/pages/SignupHandler/index.js

```diff
@@ -121,7 +120,17 @@ const SignupHandler = props => {
 
   const handleSignupSubmit = () => {
     const { dispatch } = props;
-    dispatch(routerRedux.push('/login'));
+    dispatch({
+      type: 'auth/registerUser',
+      payload: {
+        firstName,
+        lastName,
+        // TODO: Will the username be the email?
+        username: firstName.toLowerCase(),
```

Comment:
A while ago I wrote up issue #2182 on the server side, "Reconcile dashboard and server identity management"; we really need to decide exactly what data we're going to manage for a user, what's allowed to change, and how to represent ownership in the Elasticsearch documents using an immutable representation of a user.

One option is to drop "username" and internally use the Postgresql row ID; I don't really like that because it's harder to read. Email isn't good because we definitely want users to be able to change their contact email over time. We might require a text username, make that immutable, and use that internally ... TBD. While "first name" probably isn't a good longer term solution (although, miraculously, our team doesn't have duplicate first names), it's reasonable for now. Another would be, as you say in the comment, to use email. I wouldn't worry about it for this PR, though.

Reply:
> One option is to drop "username" and internally use the PostgreSQL row ID; I don't really like that because it's harder to read.

Not sure what you mean by harder to read, but in the dashboard we should use email as the handle, and an immutable ID (database row ID).  Email should be verified before used, and once verified, then it has to be unique anyways.  The user can change it, or we can support adding multiple emails per ID.

> Email isn't good because we definitely want users to be able to change their contact email over time.

That is possible using a database ID.  The user can add an email or change it.

> We might require a text username, make that immutable, and use that internally ... TBD.

I am not in favor of this because email has to be unique anyways, and we always have a database ID.

> While "first name" probably isn't a good longer term solution (although, miraculously, our team doesn't have duplicate first names), it's reasonable for now. Another would be, as you say in the comment, to use email. I wouldn't worry about it for this PR, though.

I agree, let's get this PR to land, we can change to email in a future PR.